### PR TITLE
[RFR] First integration tests for Nuage provider

### PR DESCRIPTION
**Pull request by miha-plesko**
With this commit we implement the very first pair of tests for Nuage provider. First test just opens up a form for creating new Nuage provider, clicks "Cancel" and then asserts specific popup content. Second test inputs valid credentials, asserts correct inventory (using wrapanpi) and then deletes provider.

Depends on: https://github.com/ManageIQ/wrapanapi/pull/208

### CODE REVIEW

**Comment by miha-plesko** [comment]
Traceback: 
[traceback.txt](https://github.com/ManageIQ/integration_tests/files/1647164/traceback.txt)

**Comment by dajoRH** [comment]
Lint report for commit b62f6fb534844065ce7c5c013748ff95dcf671ab:

`cfme/tests/networks/test_network_providers.py`:
- :red_circle: Line 4:1: **F401** *'uuid' imported but unused*
- :red_circle: Line 9:1: **F401** *'cfme.utils.update.update' imported but unused*

Please, rectify these issues :smirk: .


**No commit flaws detected**.

*CFME QE Bot*

**Comment by mfalesni** [comment]
@miha-plesko no worries, as long as it is marked WIP, nobody will pay attention :)

**Comment by miha-plesko** [comment]
Hey guys, may I ask for a pair of eyes on this PR? I think it's ready to be reviewed and merged before we continue with new tests for Nuage provider.

**Comment by dajoRH** [comment]
Lint report for commit d81343ab608ca0968210f55ae52d5e112d8945c0:

`cfme/tests/networks/test_network_providers.py`:
- :red_circle: Line 4:1: **F401** *'uuid' imported but unused*
- :red_circle: Line 9:1: **F401** *'cfme.utils.update.update' imported but unused*

Please, rectify these issues :smirk: .


**No commit flaws detected**.

*CFME QE Bot*

File: cfme/tests/networks/test_network_providers.py [hunk 1]
```diff
@@ -0,0 +1,60 @@
+# -*- coding: utf-8 -*-
+# pylint: disable=E1101
+# pylint: disable=W0621
```
Comment (quarckster) [inline cfme/tests/networks/test_network_providers.py]: **REQUIRED**
Please remove this pylint commands.

File: cfme/tests/networks/test_network_providers.py [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+    except MoveTargetOutOfBoundsException:
+        # TODO: Remove once fixed 1475303
+        prov.create(cancel=True, validate_credentials=False)
+    view = prov.browser.create_view(NetworkProvidersView)
+    view.flash.assert_success_message('Add of Network Manager was cancelled by the user')
+
+
+@pytest.mark.usefixtures('has_no_networks_providers')
+@test_requirements.discovery
+def test_provider_add_with_bad_credentials(provider):
```
Comment (quarckster) [inline cfme/tests/networks/test_network_providers.py]: **REQUIRED**
```python
def test_provider_add_with_bad_credentials(provider, has_no_networks_providers):
```

File: cfme/tests/networks/test_network_providers.py [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+    default_credentials.secret = 'notyourday'
+
+    with error.expected(flash):
+        provider.create(validate_credentials=True)
+
+
+@pytest.mark.smoke
+@pytest.mark.usefixtures('has_no_networks_providers')
+@test_requirements.discovery
+def test_provider_create_new_then_delete(provider):
```
Comment (quarckster) [inline cfme/tests/networks/test_network_providers.py]: **REQUIRED**
```python
def test_provider_create_new_then_delete(provider, has_no_networks_providers):
```

File: cfme/tests/networks/test_network_providers.py [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+
+from cfme.utils import error
+from cfme.common.provider_views import NetworkProvidersView
+from cfme import test_requirements
+from cfme.networks.provider.nuage import NuageProvider, NetworkProvider
+
+pytestmark = [pytest.mark.provider([NetworkProvider], scope="module")]
+
+
+@test_requirements.discovery
```
Comment (quarckster) [inline cfme/tests/networks/test_network_providers.py]: **REQUIRED**
```python
pytestmark = [
    test_requirements.discovery
    pytest.mark.ignore_stream('5.8')
]

File: cfme/common/provider.py [hunk 5]
```diff
@@ -187,27 +187,35 @@ def create(self, cancel=False, validate_credentials=True, check_existing=False,
...
-                    raise AssertionError("Provider wasn't added. It seems form isn't accurately"
-                                         " filled")
+                self._post_create_checks(main_view, add_view)
 
         if validate_inventory:
             self.validate()
 
         return created
 
+    def _post_create_checks(self, main_view, add_view=None, cancel=False):
```
Comment (quarckster) [inline cfme/common/provider.py]: **REQUIRED**
It's not needed. We plan to assert flash messages in the tests. Only `flash.assert_no_error()` will be kept.

**Comment by quarckster** [comment]
I'm waiting for [ManageIQ/wrapanapi#208](https://github.com/ManageIQ/wrapanapi/pull/208) to be merged than PRT can be run.

**Comment by miha-plesko** [comment]
@quarckster is there any chance we could merge this PR with `_post_create_checks` function introduced and do the refactoring later as suggested by you in [this](https://github.com/ManageIQ/integration_tests/pull/6499#discussion_r164097424) comment? I don't find it too reasonable to wait for refactoring to just happen..

**Comment by miha-plesko** [comment]
Hey, so the dependant [wrapanapi PR](https://github.com/ManageIQ/wrapanapi/pull/208) has been merged, could we push this forward now?

**Comment by quarckster** [comment]
@psav

**Comment by miha-plesko** [comment]
@psav I've just tested if this still works, and yes, it does. Should be ready to go.

**Comment by miha-plesko** [comment]
@psav I've applied your comments (as discussed on gitter) and repushed. Is there anything else I can do to improve the code?

**Pull request merged**